epsagon/events/httplib2.py

```diff
@@ -0,0 +1,146 @@
+"""
+httplib2 events module.
+"""
+
+from __future__ import absolute_import
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse
+import traceback
+from uuid import uuid4
+import simplejson
+
+from epsagon.utils import add_data_if_needed
+from ..trace import tracer
+from ..event import BaseEvent
+from ..wrappers.http_filters import is_blacklisted_url
+from ..utils import update_api_gateway_headers
+
+
+class Httplib2Event(BaseEvent):
+    """
+    Represents base gttplib2 event.
+    """
+
+    ORIGIN = 'httplib2'
+    RESOURCE_TYPE = 'http'
+
+    #pylint: disable=W0613
+    def __init__(self, wrapped, instance, args, kwargs, start_time, response,
+                 exception):
+        """
+        Initialize.
+        :param wrapped: wrapt's wrapped
+        :param instance: wrapt's instance
+        :param args: wrapt's args
+        :param kwargs: wrapt's kwargs
+        :param start_time: Start timestamp (epoch)
+        :param response: response data
+        :param exception: Exception (if happened)
+        """
+
+        super(Httplib2Event, self).__init__(start_time)
+        self.event_id = 'httplib2-{}'.format(str(uuid4()))
+
+        # Params can be set via args or kwargs.
```

Comment:
a better approch would be to define an args unroller:
```
def unroller(uri='N/A', method='N/A', body=None, headers=None):
    return uri, method, body, headers

uri, method, body, headers = unroller(*args, **kwargs)
```

Reply:
Awesome! Done